File: src/examples/node/pickups/create-pickup.js
const Easypost = require('@easypost/api');
const dotenv = require('dotenv');

dotenv.config({ path: '/Users/jhammond/git/easypost/easypost-tools/.env' });
const api = new Easypost(process.env.EASYPOST_TEST_API_KEY);

const address = 'adr_123...';
const shipment = 'shp_123...';

const pickup = new api.Pickup({
  address,
  shipment,
  reference: 'my-first-pickup',
  min_datetime: '2020-11-20 10:30:00',
  max_datetime: '2020-11-21 10:30:00',
  is_account_address: false,
  instructions: 'Special pickup instructions',
  carrier_accounts: ['ca_123...']
});

pickup.save().then(console.log).catch(console.log);

File: src/examples/python/pickups/create_pickup.py
import os

import easypost
from dotenv import load_dotenv

load_dotenv()
easypost.api_key = os.getenv('EASYPOST_TEST_API_KEY')

pickup = easypost.Pickup.create(
    address={"id": "adr_123"},
    shipment={"id": "shp_123"},
    reference="my-first-pickup",
    min_datetime="2022-07-21 10:30:00",
    max_datetime="2022-07-22 10:30:00",
    is_account_address=False,
    instructions="Special pickup instructions",
)

print(pickup)
